src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -496,12 +496,21 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                             win32ResourceStreamOpt,
                                             diagnosticBag,
                                             cancellationToken);
+                                    }
 
-                                        if (success)
-                                        {
-                                            compilation.ReportUnusedImports(null, diagnosticBag, cancellationToken);
-                                        }
+                                    var usingDiagnostics = DiagnosticBag.GetInstance();
+
+                                    // call ReportUnusedImports unconditionally
+                                    // since that also signals the event queue about completion of compiling.
+                                    compilation.ReportUnusedImportsAndFinishCompiling(null, usingDiagnostics, cancellationToken);
```

Comment:
I thought about this, but then if ReportUnusedImports call the CompleteTree as well we call it twice?
I guess, I can make ReportUnusedImports to not call CompleteTree.

Will try.